kong/utils.go

```diff
@@ -208,12 +208,53 @@ func getConfigSchema(schema gjson.Result) (gjson.Result, error) {
 	return schema, fmt.Errorf("no 'config' field found in schema")
 }
 
+func traverseConfigMap(currentConfigMap map[string]interface{}, path []string) (interface{}, error) {
+	if len(path) == 0 {
+		return nil, nil
+	}
+
+	pathElement := path[0]
+	value, ok := currentConfigMap[pathElement]
+	if !ok {
+		return nil, fmt.Errorf("key %q not found in map", pathElement)
+	}
+
+	switch v := value.(type) {
+	case map[string]interface{}:
+		// Traversing the map recursively, dissecting the path each time
+		return traverseConfigMap(v, path[1:])
+	default:
+		return v, nil
+	}
+}
+
+func backfillResultConfigMap(res Configuration, path []string, configValue interface{}) error {
+	// Traverse the map to the second-to-last level
+	for i, p := range path {
+		if i == len(path)-1 {
+			// Last element in the path, update the value
+			res[p] = configValue
+			return nil
+		}
+		// Traverse to the next level
+		next, ok := res[p].(map[string]interface{})
+		if !ok {
+			return fmt.Errorf("backward_translation path incorrect")
```

Comment:
Let's include the offending `path` in the error message so that it's easier to debug in case this happens: 

```suggestion
			return fmt.Errorf("backward_translation path %q incorrect", p)
```